project/project.go

```diff
@@ -67,20 +68,30 @@ func (m *ProjectMeta) GetConfigs(ipfsEndpoint string) ([]*Config, error) {
 		content []byte
 		err     error
 	)
-	if u, _err := url.Parse(m.Uri); _err == nil && (u.Scheme == "http" || u.Scheme == "https") {
-		// fetch content from ipfs gateway
-		resp, _err := http.Get(m.Uri)
-		if _err != nil {
-			return nil, errors.Wrapf(err, "fetch project config failed, projectID %d, uri %s", m.ProjectID, m.Uri)
-		}
-		defer resp.Body.Close()
-
-		// TODO network error should try again
-		content, err = io.ReadAll(resp.Body)
+	u, _err := url.Parse(m.Uri)
+	if _err != nil {
+		return nil, errors.Wrapf(err, "failed to parse project url: %s", m.Uri)
 	} else {
```

Comment:
the else is not necessary

Reply:
yes, if the url was parsed incorrectly, treat it as a raw ipfs CID.